File: routes/GET/getSimilarAnime.py
from classes import animes
from flask import make_response, jsonify

def get(animeId):
    try:
        anime = animes.Anime(id=animeId)
        getSimilarAnimes = anime.getSimilarAnime()
        match getSimilarAnimes["message"]:
            case "Animes found":
                resp = make_response(jsonify(getSimilarAnimes))
                resp.status_code = 200
                return resp
            case "An error has occurred while getting the anime":
                resp = make_response(jsonify(getSimilarAnimes))
                resp.status_code = 500
                return resp
        
    except Exception as e:
        resp = make_response(jsonify({"message":"An unknown error has occurred", "error":e.args}))

File: src/routes/GET/getSimilarAnime.ts
import { Anime } from "../../classes/animes";
import type { Response, Request } from "express";

export default async function getSimilarAnimes(req: Request, res: Response){
  try {
    const animeId = Number(req.params.animeId);
    const animes =await new Anime(animeId).getSimilar();
    switch (animes.message) {
      case "no genres found":
        res.statusCode = 404;
        res.json(animes);
        break;
      case "success":
        res.statusCode = 200;
        res.json(animes);
        break;
      case "An error has occurred while getting the similar Animes":
        res.statusCode = 500;
        res.json(animes);
    }
  } catch (error: any) {
    res.statusCode = 500;
    res.json({ message: "there was an error", error: error.message });
  }
}
